jgrapht-guava/src/main/java/org/jgrapht/graph/GraphImmutableNetworkAdapter.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * (C) Copyright 2018-2018, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+import org.jgrapht.EdgeFactory;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphType;
+import org.jgrapht.util.TypeUtil;
+
+import com.google.common.graph.Graphs;
+import com.google.common.graph.ImmutableNetwork;
+import com.google.common.graph.NetworkBuilder;
+
+/**
+ * A graph adapter class using Guava's {@link ImmutableNetwork}.
+ * 
+ * @author Dimitrios Michail
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class GraphImmutableNetworkAdapter<V, E>
```

Comment:
For the immutable variants, it seems like it would be easier to just our our AsUnmodifiable wrapper, but I haven't thought it through, so if there's an obvious reason not to, just ignore this comment.